src/lib.rs

```diff
@@ -56,12 +57,28 @@ macro_rules! impl_array {
                 mut topic_parts: core::iter::Peekable<core::str::Split<char>>,
                 value: &[u8],
             ) -> Result<(), Error> {
-                if topic_parts.peek().is_some() {
-                    return Err(Error::NameTooLong);
+                if let Some(next) = topic_parts.next() {
+                    // Parse what should be the index value
+                    let i: usize = serde_json_core::from_str(next)?.0;
```

Comment:
Instead of ? on this error, can we catch it and return a `miniconf::Error::InvalidIndex` or something similar? I think a deserialization error would be less intuitive if someone forgot the index.